doc/developer/design/20210311_prometheus_source.md

```diff
@@ -0,0 +1,144 @@
+# Summary
+
+This design doc introduces a way to import data from prometheus metrics endpoints into materialized views that can be inspected with SQL commands.
+
+# Goals
+
+* Allow our users to read & make decisions based on our metrics data without setting up a whole metrics ingestion/presentation pipeline.
+
+* Let administrators inspect that metrics data (as much as metrics' attached data allows) together with the other available data we present in system tables.
+
+* Retain a long-enough history of metrics values to allow making decisions based on how the metrics values change (but short enough to not consume unbounded memory).
```

Comment:
I don't believe that this is a current goal -- and your comment here https://github.com/MaterializeInc/materialize/issues/1696#issuecomment-796949699 suggests that originally you felt the same way?

I think that this in particular adds a lot of complexity that we should strive to avoid for v0.

Reply:
I've been wavering between the two: Retaining some history would make it easier to aggregate things, but on the other hand, make it a bit harder to query for the "current" value.

I feel it's probably better to have some (minor) access to historic values (so users can use query tools to chart the progression of a metric) than to throw that info away, and given that we're making timely dataflows from the information, it felt like this information would be easy to retain & process with (especially since we need a timestamp on a source anyway).

I'm probably missing the source of complexity that you're seeing - what do you think would make this difficult?

Reply:
Perhaps we should make this goal optional or a stretch goal? I agree that it adds complexity and it would be good to know if we consider "values over time" a hard requirement.